color_refinement.py

```diff
@@ -78,6 +79,60 @@ def color_refine(coloring: "Coloring") -> "Coloring":
     return coloring
 
 
+def fast_color_refine(graph: "Graph"):
+    # takes initial coloring
+    qlist = DoubleLinkedList()
+    color_map = {}
+    neigh_map = {}
+    #TODO: Replace this with better things, because it's now just used to get the amount of colors
+    # because I did not figure out how to do this with the Coloring class yet
+    # Neigh_map might not be neccessary
+    for v in graph.vertices:
+        if v.colornum not in color_map.keys():
+            color_map[v.colornum] = []
+        color_map[v.colornum].append(v)
+        if v not in neigh_map.keys():
+            neigh_map[v] = None
+        neigh_map[v] = v.neighbours
+    first_color = sorted(list(color_map.keys()))[0]
+    qlist.append(Node(first_color))
+    #TODO: Create a loop with the queue
+    class_list = {}
+    for v in graph.vertices:
+        count = 0
+        if v.colornum is not qlist._first_node:
+            for x in neigh_map.get(v):
+                if x.colornum is qlist._first_node:
+                    count += 1
+        if v.colornum not in class_list.keys():
+            class_list[v.colornum] = []
+        class_list[v.colornum].append((v, count))
+    for c in class_list.keys():
+        to_split = {}
+        for x, (y, w) in enumerate(class_list[c]):
```

Comment:
Graag deze variablen een wat logischere naam geven, nu is het moeilijk te begrijpen wat ze zijn.

Reply:
Wat is x? Want y is een vertex en w een count, maar wat is x dan? Gelijk aan c?

Reply:
Klopt, dit moet beter. Ik moet even kijken of dit makkelijker kan want op dit moment is mijn datastructuur een beetje een zooitje waardoor het zo ingewikkeld wordt.